lib/screens/login/login_view_model.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+final loginViewModelProvider =
+    StateNotifierProvider.autoDispose<LoginViewModel, void>((_) {
+  return LoginViewModel();
+});
+
+class LoginViewModel extends StateNotifier<void> {
+  LoginViewModel() : super([]);
+
+  bool validateEmail(String? email) {
+    // Just use a simple rule, no fancy Regex!
+    return !(email == null || email.isEmpty || !email.contains('@'));
```

Comment:
I think `!email.contains('@')` 's already covered the empty case
```suggestion
    return !(email == null || !email.contains('@'));
```